### PR TITLE
[FIX] Broken navigation from Directory to Omnichannel contact center

### PR DESCRIPTION
**Pull request by Aman-Maheshwari**
- [x] I have read the Contributing Guide.
  - [x] I have signed the CLA.
  - [x] Lint and unit tests pass locally with my changes
  - [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
  - [ ] I have added necessary documentation (if applicable)
  - [ ] Any dependent changes have been merged and published in downstream modules

## Proposed changes (including videos or screenshots)

Fixes the broken navigation from directory to omnichannel contact center.

## Before

https://user-images.githubusercontent.com/50165440/140654685-d2281491-fe37-4b53-bafc-2f3775142595.mp4

## After

https://user-images.githubusercontent.com/50165440/140654638-38253c1e-3983-4fed-85fb-e45d285a35a8.mp4

## Issue(s)

#23534 

## Steps to test or reproduce

Mentioned in the issue

### CODE REVIEW

File: client/views/directory/DirectoryPage.js [hunk 1]
```diff
@@ -17,10 +17,10 @@ function DirectoryPage() {
 	const [routeName] = useCurrentRoute();
 	const tab = useRouteParameter('tab');
 	const directoryRoute = useRoute('directory');
-
+	const navRoute = useRoute(routeName);
 	useEffect(() => {
 		if (routeName !== 'directory') {
-			return;
+			return navRoute.push();
```
Comment (murtaza98) [inline client/views/directory/DirectoryPage.js:23]: @Aman-Maheshwari  FYI: There's a warning being introduced here. Could you please resolve it? Thanks :smiley: 

image

Comment (Aman-Maheshwari) [reply client/views/directory/DirectoryPage.js:23]: sure :+1:

Comment (Aman-Maheshwari) [reply client/views/directory/DirectoryPage.js:23]: @murtaza98 can you please review it ? I have resolved the warning. Thank you :smile:

**Review by tiagoevanp — COMMENTED** [review]
The root of the problem is not solved by this PR.
Different solution in:  https://github.com/RocketChat/Rocket.Chat/pull/23691

**Pull request closed without merging**